pyeapi/api/acl.py

```diff
@@ -152,5 +215,90 @@ def remove_entry(self, name, seqno):
         return self.configure(cmds)
 
 
+class ExtendedAcls(EntityCollection):
+
+    entry_re = re.compile(r'(\d+)'
+                          r'(?: ([p|d]\w+))'
+                          r'(?: (\w+|\d+))'
+                          r'(?: (any))?'
+                          r'(?: (host))?'
+                          r'(?: ([0-9]+(?:\.[0-9]+){3}))?'
+                          r'(?:/([0-9]{1,2}))?'
+                          r'(?: (eq [\w-]+))?'
+                          r'(?: (any))?'
+                          r'(?: (host))?'
+                          r'(?: ([0-9]+(?:\.[0-9]+){3}))?'
+                          r'(?:/([0-9]{1,2}))?'
+                          r'(?: ([0-9]+(?:\.[0-9]+){3}))?'
+                          r'(?: (eq [\w-]+))?'
+                          r'(?: (.+))?')
+
+    def get(self, name):
+        config = self.get_block('ip access-list %s' % name)
+        if not config:
+            return None
+        resource = dict(name=name, type='extended')
+        resource.update(self._parse_entries(config))
+        return resource
+
+    def _parse_entries(self, config):
+        entries = dict()
+        for item in re.finditer(r'\d+ [p|d].*$', config, re.M):
+            match = self.entry_re.match(item.group(0))
+            entry = dict()
+            entry['action'] = match.group(2)
+            entry['protocol'] = match.group(3)
+            entry['srcaddr'] = match.group(6) or 'any'
+            entry['srclen'] = match.group(7)
+            entry['srcport'] = match.group(8)
+            entry['dstaddr'] = match.group(11) or 'any'
+            entry['dstlen'] = match.group(12)
+            entry['dstport'] = match.group(14)
+            entry['other'] = match.group(15)
+            entries[match.group(1)] = entry
+        return dict(entries=entries)
+
+    def create(self, name):
+        return self.configure('ip access-list %s' % name)
+
+    def delete(self, name):
+        return self.configure('no ip access-list %s' % name)
+
+    def default(self, name):
+        return self.configure('default ip access-list %s' % name)
+
+    def update_entry(self, name, seqno, action, protocol, srcaddr,
+                     srcprefixlen, dstaddr, dstprefixlen, log=False):
+        cmds = ['ip access-list %s' % name]
+        cmds.append('no %s' % seqno)
+        entry = '%s %s %s %s/%s %s/%s' % (seqno, action, protocol, srcaddr,
+                                          srcprefixlen, dstaddr, dstprefixlen)
```

Comment:
Currently I've remained consistent with how standard ACLs are handling the host keyword for updated/new entries. The user is required to put in 0 for the host prefix length and EOS will translate it to the host keyword.